src/vm/eventpipesession.cpp

```diff
@@ -89,4 +86,157 @@ EventPipeSessionProvider* EventPipeSession::GetSessionProvider(EventPipeProvider
     return m_pProviderList->GetSessionProvider(pProvider);
 }
 
+void EventPipeSession::WriteAllBuffersToFile(
+    EventPipeFile &fastSerializableObject,
+    EventPipeConfiguration &configuration,
+    LARGE_INTEGER stopTimeStamp)
+{
+    CONTRACTL
+    {
+        THROWS;
+        GC_NOTRIGGER;
+        MODE_ANY;
+        PRECONDITION(EventPipe::IsLockOwnedByCurrentThread());
+    }
+    CONTRACTL_END;
+
+    m_pBufferManager->WriteAllBuffersToFile(&fastSerializableObject, configuration, stopTimeStamp);
+}
+
+bool EventPipeSession::WriteEvent(
+        Thread *pThread,
+        EventPipeEvent &event,
+        EventPipeEventPayload &payload,
+        LPCGUID pActivityId,
+        LPCGUID pRelatedActivityId,
+        Thread *pEventThread,
+        StackContents *pStack)
+{
+    return m_pBufferManager->WriteEvent(pThread, *this, event, payload, pActivityId, pRelatedActivityId);
```

Comment:
I assume there is some TODO work in order to filter events to specific sessions that enabled them. If that was already occuring somewhere I didn't catch it : )

Reply:
You are right, there is an implicit TODO here (I added it to the PR description but forgot to add it here).